app/controllers/death_records_controller.rb

```diff
@@ -174,11 +169,76 @@ def request_edits
 
   # Return a list of users given a role
   def users_by_role
-    render :json => User.with_role(params[:role]).map { |user| "#{user.email} (#{NameHelper.pretty_user_name(user)})" }
+    render json: User.with_role(params[:role]).map { |user| "#{user.email} (#{NameHelper.pretty_user_name(user)})" }
+  end
+
+  # Retrieve all DeathRecords currently owned by this user.
+  def owned_death_records
+    length = params[:length].to_i
+    page = params[:start].to_i == 0 ? 1 : (params[:start].to_i / length) + 1
+    draw = params[:draw].to_i
+    search = params[:search][:value] unless params[:search].nil?
+    if current_user.admin?
+      records = DeathRecord.all
+    else
+      records = current_user.owned_death_records.where(abandoned: false)
+    end
+    if search.present?
+      filtered = records.where('lower(name) like ?', "%#{search.downcase}%")
+    else
+      filtered = records
+    end
+    # NOTES:
+    # 'data' is the array of records
+    # 'recordsTotal' is the TOTAL number of records
+    # 'recordsFiltered' is the number of records AFTER filtering (but not after pagination!)
+    render json: {data: filtered.page(page).per(length), draw: draw, recordsTotal: records.count, recordsFiltered: filtered.count}
+  end
+
+  # Retrieve all DeathRecords created by this user (that aren't currently
+  # owned by this user).
+  # TODO: This should be made more efficient at some point. It will only send
+  # the number or records required for the front end (paginated), but there
+  # are still ways this could be improved on the backend.
+  def transferred_death_records
+    length = params[:length].to_i
+    page = params[:start].to_i == 0 ? 1 : (params[:start].to_i / length) + 1
+    draw = params[:draw].to_i
+    search = params[:search][:value] unless params[:search].nil?
+    transferred = []
+    total = 0
+    current_user.step_histories.includes(:death_record).each do |history|
```

Comment:
Is this loop because we don't track which User had ownership of a death record? Maybe we create a table that maps OwnerID -> DeathRecordID? I could be missing something.